disorder-core/src/Disorder/Core/Gen.hs

```diff
@@ -15,3 +17,10 @@ vectorOfSize min' max' gen =
 chooseSize :: Int -> Int -> Gen Int
 chooseSize min' max' =
   sized (return . min max' . max min')
+
+-- | from a generator return a generator that will generate Nothing
+--   half the time and Just half the time
+maybeGen :: Gen a -> Gen (Maybe a)
+maybeGen g = do
+  b <- elements [True, False]
```

Comment:
and what about sizes? it's not as big a deal because Maybe isn't used recursively very often, but I'd do something like:
```
sized $ s
 -> if   s == 0
     then pure Nothing
     else resize (s-1) (oneof [Just <$> g, pure Nothing])
```